tiledb/sm/query/sparse_index_reader_base.cc

```diff
@@ -337,9 +388,36 @@ Status SparseIndexReaderBase::compute_tile_bitmaps(
     return Status::Ok();
   }
 
+  // Compute parallelization parameters.
+  uint64_t num_cell_ranges = 1;
+  const auto num_threads = storage_manager_->compute_tp()->concurrency_level();
+  if (result_tiles->size() < num_threads) {
+    // Ceil the divition between thread_num and tile_num.
+    num_cell_ranges = 1 + ((num_threads - 1) / result_tiles->size());
+  }
+
+  // Perforance runs have shown that running multiple parallel_for's has a
+  // measurable performance impact. So only pre-allocate tile bitmaps if we
+  // are going to run multiple cell ranges.
+  if (num_cell_ranges != 1) {
+    // Resize bitmaps to process for each tiles in parallel.
+    auto status = parallel_for(
```

Comment:
I am a little unclear on this. Are you suggesting that parallel bitmap allocation boosts performance?

Reply:
I was thinking it would, but upon reading a little on this further, it might not. I'll run more tests...